lldb/test/API/functionalities/data-formatter/type-recognizer/value-object-cast/main.cpp

```diff
@@ -0,0 +1,149 @@
+enum class CastTarget {
+  A_1,
+  A_2,
+
+  B_1,
+  B_2,
+  B_3,
+  B_4,
+
+  C_1,
+  C_2,
+  C_3,
+  C_4,
+};
+
+/// Non-Virtual Inheritance:
+/// --------------------------
+///
+///  [A_1]  [A_1] [A_2]
+///    |       \   /
+///  [B_1]     [B_2]
+///      \     /
+///       [C_1]
+///
+///
+/// Virtual Inheritance:
+/// --------------------------
+///
+///       [A_2]
+///      /     \
+///  [B_3]     [B_4]
+///      \     /
+///       [C_2]
+///
+///
+///  [A_1]    [A_1] [A_2]
+///    |         \   /
+///  [B_1]       [B_2]
+///    |           |
+///   virt       virt
+///     \         /
+///    [--- C_3 ---]
+///
+///
+///  [A_1]    [A_1] [A_2]
+///    |         \   /
+///  [B_1]       [B_2]
+///    |           |
+///    |         virt
+///     \         /
+///    [--- C_4 ---]
+///
+
+//===-- [Base classes] ----------------------------------------------------===//
+
+struct A_1 {
+  virtual void foo() {}
+  CastTarget a_1_target{CastTarget::A_1};
+  CastTarget a_1_member{CastTarget::A_1};
+};
+
+struct A_2 {
+  virtual void bar() {}
+  CastTarget target{CastTarget::A_2};
+  CastTarget a_2_member{CastTarget::A_2};
+};
+
+//===-- [Intermediate classes] --------------------------------------------===//
+
+struct B_1 : A_1 {
+  CastTarget b_1_member{CastTarget::B_1};
+};
+
+struct B_2 : A_2, A_1 {
+  CastTarget b_2_member{CastTarget::B_2};
+};
+
+struct B_3 : virtual A_2 {
+  CastTarget b_3_member{CastTarget::B_3};
+};
+
+struct B_4 : virtual A_2 {
+  CastTarget b_4_member{CastTarget::B_4};
+};
+
+//===-- [Derived classes] -------------------------------------------------===//
+
+struct C_1 : B_1, B_2 {
+  CastTarget c_1_member{CastTarget::C_1};
+};
+
+struct C_2 : B_3, B_4 {
+  CastTarget c_2_member{CastTarget::C_2};
+};
+
+struct C_3 : virtual B_1, virtual B_2 {
+  CastTarget c_3_member{CastTarget::C_3};
+};
+
+struct C_4 : B_1, virtual B_2 {
+  CastTarget c_4_member{CastTarget::C_4};
+};
+
+//===-- [Function section] ------------------------------------------------===//
+
+template <class T> void cast_func(T *input_var) {
+  [[maybe_unused]] int o{0}; // Break here in cast_func().
+}
+
+int main() {
+  // Prepare objects
```

Comment:
No need to first prepare a bunch of independent objects, and then use them. Move them to their respective test cases